src/backend/core/models.py

```diff
@@ -367,6 +368,54 @@ def _get_abilities(self, resource, user):
         }
 
 
+class DocumentQuerySet(models.QuerySet):
+    """
+    Custom queryset for the Document model, providing additional methods
+    to filter documents based on user permissions.
+    """
+
+    def readable(self, user):
+        """
+        Filters the queryset to return documents that the given user has
+        permission to read.
+
+        :param user: The user for whom readable documents are to be fetched.
+        :return: A queryset of documents readable by the user.
+        """
+        if user.is_authenticated:
+            return self.filter(
+                models.Q(accesses__user=user)
+                | models.Q(accesses__team__in=user.teams)
+                | ~models.Q(link_reach=LinkReachChoices.RESTRICTED)
+            )
+
+        return self.filter(models.Q(link_reach=LinkReachChoices.PUBLIC))
+
+
+class DocumentManager(models.Manager):
+    """
+    Custom manager for the Document model, enabling the use of the custom
+    queryset methods directly from the model manager.
+    """
+
+    def get_queryset(self):
+        """
+        Overrides the default get_queryset method to return a custom queryset.
+
+        :return: An instance of DocumentQuerySet.
+        """
+        return DocumentQuerySet(self.model, using=self._db)
+
+    def readable(self, user):
+        """
+        Filters documents based on user permissions using the custom queryset.
+
+        :param user: The user for whom readable documents are to be fetched.
+        :return: A queryset of documents readable by the user.
+        """
+        return self.get_queryset().readable(user)
```

Comment:
I don't understand this. Don't you want to consider using https://docs.djangoproject.com/en/5.1/topics/db/managers/#creating-a-manager-with-queryset-methods

Like `objects = DocumentQuerySet.as_manager()`